liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/BasicSupport.java

```diff
@@ -198,17 +189,40 @@ protected void init() throws MojoExecutionException, MojoFailureException {
                 installType = InstallType.FROM_FILE;
                 installDirectory = checkServerHome(assemblyArchive);
                 log.info(MessageFormat.format(messages.getString("info.variable.set"), "installDirectory", installDirectory));
-            } else if (assemblyArtifact != null) {
+            } else if(install != null) {
+                installType = InstallType.FROM_ARCHIVE;
+                installDirectory = new File(assemblyInstallDirectory, "wlp");
+                log.info(MessageFormat.format(messages.getString("info.variable.set"), "installDirectory", installDirectory));
+            }
+            else { // default to install from runtime artifact
+                assemblyArtifact.setType("zip");
```

Comment:
Couldn't assemblyArtifact be null at this point?

Reply:
No, it should never be null.